sdk/samples/iot/paho_iot_pnp_component_sample.c

```diff
@@ -0,0 +1,626 @@
+
+  // Applications call az_iot_hub_client_properties_get_next_component_property to enumerate
+  // properties received.
+  while (az_result_succeeded(
+      rc = az_iot_hub_client_properties_get_next_component_property(
+          &hub_client, &jr, response_type, AZ_IOT_HUB_CLIENT_PROPERTY_WRITEABLE, &component_name)))
+  {
+    if (rc == AZ_OK)
+    {
+      // Once we have received a property, determine which component the property is intended for
+      // and route to the intended property component handler.  The component handler
+      // takes over from this point, for both parsing the request and responding to IoT Hub.
+      if (az_span_is_content_equal(component_name, thermostat_1_name))
+      {
+        pnp_thermostat_process_property_update(
+            &thermostat_1, &hub_client, mqtt_client, &jr, version);
+      }
+      else if (az_span_is_content_equal(component_name, thermostat_2_name))
+      {
+        pnp_thermostat_process_property_update(
+            &thermostat_2, &hub_client, mqtt_client, &jr, version);
+      }
+      else
+      {
+        char const* const log_message = "Failed to process property update";
+
+        // Only the thermostat component supports writeable properties;
+        // the models for DeviceInfo and the temperature controller do not.
+        IOT_SAMPLE_LOG_ERROR("Received property update for an unsupported component");
+
+        // We do NOT report back an error for an unknown property to IoT Hub,
+        // but we do need to skip past the JSON part of the body to continue reading.
+        IOT_SAMPLE_EXIT_IF_AZ_FAILED(az_json_reader_next_token(&jr), log_message);
+        IOT_SAMPLE_EXIT_IF_AZ_FAILED(az_json_reader_skip_children(&jr), log_message);
+        IOT_SAMPLE_EXIT_IF_AZ_FAILED(az_json_reader_next_token(&jr), log_message);
+      }
+    }
+    else
+    {
+      IOT_SAMPLE_LOG_ERROR("Failed to update a property: az_result return code 0x%08x.", rc);
+      exit(rc);
+    }
+  }
+}
+
+// handle_device_property_message handles incoming properties from Azure IoT Hub.
+static void handle_device_property_message(
+    MQTTClient_message const* receive_message,
+    az_iot_hub_client_properties_response const* property_response)
+{
+  az_span const message_span
+      = az_span_create((uint8_t*)receive_message->payload, receive_message->payloadlen);
+
+  // Invoke appropriate action per response type (3 types only).
+  switch (property_response->response_type)
+  {
+    // A response from a property GET publish message with the property document as a payload.
+    case AZ_IOT_HUB_CLIENT_PROPERTIES_RESPONSE_TYPE_GET:
+      IOT_SAMPLE_LOG("Message Type: GET");
+      (void)process_device_property_message(message_span, property_response->response_type);
+      break;
+
+    // An update to the desired properties with the properties as a payload.
+    case AZ_IOT_HUB_CLIENT_PROPERTIES_RESPONSE_TYPE_DESIRED_PROPERTIES:
```

Comment:
just adding a note here as you mentioned about changing this name to "writeable properties" or something like that

Reply:
ack - I'll get this in 1st and then I'll turn attention again to any cleanup in core headers.